src/errors/base-error.ts

```diff
@@ -22,9 +33,33 @@ export interface CommonErrorProperties {
  * This means that errors can be accessed using `Sequelize.ValidationError`
  */
 abstract class BaseError extends Error {
-  constructor(message?: string) {
-    super(message);
+  declare cause: Error | undefined;
+
+  get parent(): this['cause'] {
+    useErrorCause();
+
+    return this.cause;
+  }
+
+  get original(): this['cause'] {
+    useErrorCause();
+
+    return this.cause;
+  }
+
+  constructor(message?: string, options?: ErrorOptions) {
+    // TODO [>=2023-04-30]: remove this ts-ignore (Sequelize 8)
+    // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error -- Supported in TS 4.6, not before
+    // @ts-ignore
+    super(message, options);
     this.name = 'SequelizeBaseError';
+
+    if (!supportsErrorCause) {
+      // TODO [>=2023-04-30]:
+      //  Once all supported node versions have support for Error.cause (added in node 16), delete this line:
```

Comment:
```suggestion
      //  Once all supported node versions have support for Error.cause (added in Node 16.9.0), delete this line:
```
As per my earlier comment